pkg/pillar/types/sriovtypes.go

```diff
@@ -0,0 +1,181 @@
+// Copyright (c) 2022 Zededa, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package types
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+
+	uuid "github.com/satori/go.uuid"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	NicLinuxPath     = "/sys/class/net/"
+	NumvfsDevicePath = "/device/sriov_numvfs"
+	VfCountFieldName = "sriov-vf-count"
+)
+
+// Must match fields of EthVF in devcommon.proto
+type EthVF struct {
+	Index        uint8
+	PciLong      string // BFD notation
+	Iface        string
+	UsedByUUID   uuid.UUID
+	Mac          net.HardwareAddr
+	VlanId       uint16
+	Logicallabel string
+}
+
+func (pf *EthVF) fromAdapter(rhs EthVF) {
+	if rhs.Mac != nil {
+		pf.Mac = rhs.Mac
+	}
+	if rhs.Index != 0 {
+		pf.Index = rhs.Index
+	}
+	if rhs.VlanId != 0 {
+		pf.VlanId = rhs.VlanId
+	}
+}
+
+func (vf *EthVF) setupLink() error {
```

Comment:
Same comment as for the other function calling LinkSetVfHardwareAddr.

Do you need both a member function and a top-level function to set the vfvlan and vfhardwareaddr?

Reply:
Actually, we don't need it, therefore I'm removing it